org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/StandardPreferenceManager.java

```diff
@@ -135,10 +136,26 @@ public void update(Preferences preferences) {
 		if (!Objects.equals(newMavenNotCoveredExecutionSeverity, oldMavenNotCoveredExecutionSeverity)) {
 			try {
 				((MavenConfigurationImpl) getMavenConfiguration()).setNotCoveredMojoExecutionSeverity(newMavenNotCoveredExecutionSeverity);
+				for (IProject project : ProjectUtils.getAllProjects()) {
+					if (ProjectUtils.isMavenProject(project)) {
+						JavaLanguageServerPlugin.getProjectsManager().updateProject(project, true);
+					}
+				}
 			} catch (CoreException e) {
 				JavaLanguageServerPlugin.logException("failed to set not covered Maven plugin execution severity", e);
 			}
 		}
+		String newMavenDefaultMavenExecutionAction = preferences.getMavenDefaultMojoExecutionAction();
+		String oldMavenDefaultMavenExecutionAction = getMavenConfiguration().getDefaultMojoExecutionAction() == null ? null : getMavenConfiguration().getDefaultMojoExecutionAction().name();
+		if (!Objects.equals(newMavenDefaultMavenExecutionAction, oldMavenDefaultMavenExecutionAction)) {
+			PluginExecutionAction action = PluginExecutionAction.valueOf(newMavenDefaultMavenExecutionAction == null ? "ignore" : newMavenDefaultMavenExecutionAction);
+			getMavenConfiguration().setDefaultMojoExecutionAction(action);
+			for (IProject project : ProjectUtils.getAllProjects()) {
+				if (ProjectUtils.isMavenProject(project)) {
+					JavaLanguageServerPlugin.getProjectsManager().updateProject(project, true);
+				}
+			}
```

Comment:
please don't update the project twice. First determine if severity or action change, then trigger the update of maven projects

Reply:
I really wish there the preference manager was decoupled from Maven (or other build tools).
Maybe we need a BuildPreferences interface and delegate build tool specific operations to specific implementations. But that's for another PR.

Reply:
actually a simple preference listener mechanism should  help easily decouple all that

Reply:
> please don't update the projects twice. First determine if severity or action changed, then trigger the update of maven projects

Fixed.

> I really wish the preference manager was decoupled from Maven (or other build tools).
Maybe we need a BuildPreferences interface and delegate build tool specific operations to specific implementations. But that's for another PR.

Could you, please, create a new issue?